### PR TITLE
mock: fix FunctionalOptions docs

### PR DESCRIPTION
**Pull request by snirye**
## Summary
the comment about this function is not updated with the final implementation

## Changes
just the comment above the func

## Motivation
its misleading

## Related issues
some relevant links to the PR that added `FunctionalOptions`
https://github.com/stretchr/testify/pull/1023
[suggestion 1](https://github.com/stretchr/testify/pull/1023/files/3b93fae0ea161f8d0ca47dd7aa3d8a949799d261#r1065311500)
[suggestion 2](https://github.com/stretchr/testify/pull/1023/files/3b93fae0ea161f8d0ca47dd7aa3d8a949799d261#r1065309374)

### CODE REVIEW

File: mock/mock.go [hunk 1]
```diff
@@ -810,7 +810,7 @@ func (f *FunctionalOptionsArgument) String() string {
 // and the values to check of
 //
 // For example:
-// Assert(t, FunctionalOptions("[]foo.FunctionalOption", foo.Opt1(), foo.Opt2()))
+// Assert(t, FunctionalOptions(foo.Opt1("strValue"), foo.Opt2(613)))
```
Comment (dolmen) [inline mock/mock.go:813]: Please also indent this line.

File: mock/mock.go [hunk 2]
```diff
@@ -810,7 +810,7 @@ func (f *FunctionalOptionsArgument) String() string {
 // and the values to check of
```
Comment (dolmen) [inline mock/mock.go:810]: This sentence looks unfinished. It needs at last a dot, but I wonder if some word is missing.

Disclaimer: I'm not a native English speaker.

**Review by dolmen — CHANGES_REQUESTED** [review]

File: mock/mock.go (hunk 2 above)
Comment (snirye) [reply mock/mock.go:810]: what about: 
`FunctionalOptions returns an FunctionalOptionsArgument object containing the expected functional-option to check for`

im trying to keep it similar to
Lines 782 and 765

Do you think it need more detailed description?

File: mock/mock.go [hunk 3]
```diff
@@ -784,7 +784,8 @@ type IsTypeArgument struct {
 // alternative to AnythingOfType.  Used in Diff and Assert.
 //
 // For example:
-// Assert(t, IsType(""), IsType(0))
+//
+//	Assert(t, IsType(""), IsType(0))
```
Comment (snirye) [inline mock/mock.go:788]: added indentation here too

Comment (dolmen) [reply mock/mock.go:788]: This is irrelevant to the scope of this PR. Please submit this change as a separate PR.

File: mock/mock.go [hunk 4]
```diff
@@ -806,11 +807,12 @@ func (f *FunctionalOptionsArgument) String() string {
 	return strings.Replace(fmt.Sprintf("%#v", f.value), "[]interface {}", name, 1)
 }
 
-// FunctionalOptions returns an FunctionalOptionsArgument object containing the functional option type
-// and the values to check of
+// FunctionalOptions returns an FunctionalOptionsArgument object containing
+// the expected functional-options to check for
```
Comment (dolmen) [inline mock/mock.go:811]: Missing dot at the end of the sentence.

Also, add godoc link.

**Review by dolmen — CHANGES_REQUESTED** [review]

**Comment by dolmen** [comment]
Please squash commits and be more careful about the commit message.

**Comment by snirye** [comment]
Hi @dolmen , can you please review again?

**Review by dolmen — CHANGES_REQUESTED** [review]
Please fix the commit message: "remove old input args" is not helpful in the project history.

Use instead: "assert.FunctionalOptions: fix go doc"

**Review by dolmen — APPROVED** [review]

**Pull request merged**